src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -87,7 +87,11 @@ public ImmutableArray<FunctionMetadata> GetFunctionMetadata(bool forceRefresh, b
             if (forceRefresh || _servicesReset || _functionMetadataArray.IsDefaultOrEmpty)
             {
                 _functionMetadataArray = LoadFunctionMetadata(forceRefresh, includeCustomProviders, workerConfigs: workerConfigs);
-                _logger.FunctionMetadataManagerFunctionsLoaded(ApplyAllowList(_functionMetadataArray).Count());
+                // The host and worker metadata are checked in parallel, so we don't want to emit this extra logging reading the metadata
+                if (!includeCustomProviders)
```

Comment:
Currently the only place this is called with a false value is [here](https://github.com/Azure/azure-functions-host/blob/2f0a85e7ec8d7654dc2ba9254d97c57777cdcb8c/src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs#L92) right?

I don't understand why you say we don't want to log how many functions we actually load. The first log entry we write is how many function directories we've discovered (or how many worker indexing return us). The second log is how many functions actually will be active/loaded based on filtering, right? Don't we care about both?

Reply:
I was under (a potentially incorrect) assumption that we log "x functions loaded" again somewhere else? But I might have mixed that up with the double log "x functions found". I that case I think we should keep a "loaded" log and suppress the "found" log that happens twice

Reply:
That's what I remember seeing in the past - multiple "x functions loaded" lines where they disagree on the number. We should check prod logs to see

Reply:
@mathewc so the found logs should be suppressed and not the loaded logs?

These are the prod logs:

> [2023-08-14T22:29:42.083Z] Loading functions metadata
[2023-08-14T22:29:42.085Z] Reading functions metadata
[2023-08-14T22:29:42.087Z] 0 functions found
[2023-08-14T22:29:42.097Z] Reading functions metadata
[2023-08-14T22:29:42.116Z] 1 functions found
[2023-08-14T22:29:42.118Z] 0 functions loaded

[This](https://vscode.dev/github/Azure/azure-functions-host/blob/aibhandari/metadata-logging-noise/src/WebJobs.Script/Host/HostFunctionMetadataProvider.cs#L52) is where the first `0 functions found` log happens.
[This](https://vscode.dev/github/Azure/azure-functions-host/blob/aibhandari/metadata-logging-noise/src/WebJobs.Script/Host/FunctionMetadataManager.cs#L235) is where the second `1 function found` log is emitted.

If I'm understanding your comment correctly, we would not need the second `1 function found` log? This happens since `includeCustomProviders` is set to true

Reply:
Confirmed with prod it's that "found" is what is listed twice.

Does the second log always have the correct value or is it sometimes reset? i.e. 

1 functions found 
Reading functions metadata
0 functions found

I think if there is a race in some way here, it might be difficult to decide on which one to keep or get rid of?

Or do we even need to log what we found and can remove both, trusting that "loaded" is the most accurate value? But even with your example log above, thats also incorrect:

1 functions found 
0 functions loaded

- very confuse

Reply:
@liliankasem I believe the second log always has the correct value, at least when I've tested it out. I think the second value differs because of `includeCustomProviders` being true [here](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Host/FunctionMetadataManager.cs#L161). Maybe it makes sense to only include the second log then and not include the first one?

Reply:
The confusion came when we added worker indexing and this new secondary codepath, but chose to reuse the same existing FunctionMetadataProviderFunctionFound message template. Looks like this error message is emitted in 3 different places. So definitely if we see logs from all 3 of those in one host startup trace it'll be very confusing.

Perhaps we just need to update the error message so it includes details on the provider that is doing the loading so the logs can be differentiated.

Reply:
Updated the logging so that it specifies where that log came from (either HostFuncdtionMatadataProvider, WorkerFunctionMetadataProvider, or FunctionMetadataProvider)